src/test/testing_assert.hpp

```diff
@@ -4,6 +4,7 @@
 
 #include "gtest/gtest.h"
 
+#include "optimizer/abstract_syntax_tree/abstract_ast_node.hpp"
 #include "types.hpp"
```

Comment:
Not related to this line, but if I put it in a regular comment, Github won't let us have a threaded conversation:

At some point in the not so distant future, we should look into translating nodes into different physical implementations. The first thing that comes to mind is the PredicateNode. The optimizer could check which chunks have indexes that can be used. For those, the IndexScan is used. The rest use the TableScan.

The reason I'm bringing this up is because it will also require nodes (both AST and operators) to have multiple parents. No need to do anything now, but we should keep it in mind.

Reply:
This is effectively already done for Joins (`_translate_join_node()`) - do you see any need for discussion here or did you just want to mention it?

Reply:
For Joins, it only selects an implementation. That implementation is used for all chunks. I am thinking about a PredicateNode which gets translated to an IndexScan for all indexed chunks and a TableScan for all others.

No need to do anything here, just wanted to bring it up.